openmcyclus/DepleteReactor.py

```diff
@@ -8,8 +11,603 @@ class DepleteReactor(Facility):
+        the fuel out commodities.
+
+        For each of the items in the unique commodities list, 
+        if there are no requests for a given commodity name, then 
+        continue to the next commodity. Get the recipe for 
+        each commodity requested.
+
+        Looking at the composition for each commodity, 
+        if there is no composition, then move to the next commodity. 
+
+        For each request of each commodity, sum the total request 
+        for the commodity. Then create a bid for each request of 
+        each commodity.
+
+        For each material composition sum to total mass of 
+        the composition.
+
+        Add a constraint of the total quantity of the commodity available.
+        Create a bid portfolio for each request that can be met. 
+        '''
+        print("time:", self.context.time, "start get material_bids")
+        got_mats = False
+        bids = []
+        port = []
+        for commod_index, commod in enumerate(self.fuel_outcommods):
+            reqs = requests[commod]
+            print("time:", self.context.time, "reqs:", reqs)
+            if len(reqs) == 0:
+                continue
+            elif (got_mats == False):
+                all_mats = self.peek_spent()
+            if len(all_mats) == 0:
+                tot_qty = 0
+                continue
+            if commod in all_mats: 
+                mats = [all_mats[commod]]
+                
+            else:
+                mats = []
+            print("time:", self.context.time, "mats to trade matching request commod:", mats)
+            if len(mats) == 0:
+                continue 
+
+            recipe_comp = self.context.get_recipe(self.fuel_outrecipes[commod_index])
+    
+            for req in reqs:
+                tot_bid = 0
+                for jj in range(len(mats)):
+                    tot_bid += mats[jj].quantity
+                    qty = min(req.target.quantity, self.assem_size)
+                    mat = ts.Material.create_untracked(qty, recipe_comp)
+                    for kk in range(self.spent_fuel.count):
+                        bids.append({'request':req,'offer':mat})
+                    if tot_bid >= req.target.quantity:
+                         break
+            tot_qty = 0
+            for mat in mats:
+                tot_qty += mat.quantity
+        if len(bids) == 0:
+            return 
+
+        port = {'bids':bids}
+        print("time:", self.context.time, "portfolio:", port)
+        print("time:", self.context.time, "respond", len(bids), "assemblies")
+        return port
+
+    def get_material_trades(self, trades): #phase 5.1
+        '''
+        Trade away material in the spent_fuel material buffer.
+
+        Pull out the material from spent fuel inventory
+        For each trade, get the commodity name, get the 
+        composition of the trade.
+
+        Then trade the materials from the spent fuel inventory. 
+        '''
```

Comment:
Consider adding information about the parameter `trades` to this docstring and the returned `responses`